src/main/java/com/synopsys/integration/detect/Application.java

```diff
@@ -272,9 +282,9 @@ private void shutdownApplication(DetectBootResult detectBootResult, ExitCodeMana
         }
     }
 
-    private void exitApplication(ExitManager exitManager, long startTime, boolean shouldForceSuccess) {
+    private void exitApplication(ExitManager exitManager, long startTime, boolean shouldForceSuccess, Optional<AutonomousManager> autonomousManagerOptional) {
         ExitOptions exitOptions = new ExitOptions(startTime, shouldForceSuccess, SHOULD_EXIT);
-        ExitResult exitResult = exitManager.exit(exitOptions);
+        ExitResult exitResult = exitManager.exit(exitOptions, autonomousManagerOptional);
```

Comment:
Should not autonomousManagerOptional be in ExitOptions?

Reply:
It can be. I had initially placed it in there but later thought it didn't fit in that class. Either way, it doesn't seem it simplifies the code more because we need to fetch autonomousManagerOptional in `ExitManager.exit()` so that it is passed to `logDetectResults`. I'm okay with moving this to ExitOptions if that's more preferable

Reply:
Yeah I agree, putting it in exitOptions would not simplify the approach as we need it further for Debug logs. I am fine with either of both approaches.

Reply:
Ok, thanks!